File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/context/roleplay/quest/QuestActiveInformations.scala





















// Generated on 05/03/2013 10:35:05
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait QuestActiveInformations extends TypeDeserializer {
  val typeId = 381

  protected trait Type extends BaseType with Serializable {
    val typeId = QuestActiveInformations.typeId

    val questId: Short

    def serialize(buf: Buffer) {
      buf.writeShort(questId)
    }

  }

  protected def create(questId: Short): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val questId = buf.readShort()

    create(questId)
  }

}

object QuestActiveInformations extends QuestActiveInformations {
  case class TypeImpl(questId: Short) extends Type
  override def create(questId: Short) = TypeImpl(questId)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(questId: Short) = create(questId)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/QuestActiveInformations.java


// Generated on 05/08/2013 19:38:02
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class QuestActiveInformations extends NetworkType {
    public static final short TYPE_ID = 381;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public short questId;
    
    public QuestActiveInformations() { }
    
    public QuestActiveInformations(short questId) {
        this.questId = questId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(questId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        questId = buf.readShort();
        if (questId < 0)
            throw new RuntimeException("Forbidden value on questId = " + questId + ", it doesn't respect the following condition : questId < 0");
    }
    
}
